Libraries/Core/InitializeCore.js

```diff
@@ -178,7 +179,9 @@ let navigator = global.navigator;
 if (navigator === undefined) {
   global.navigator = navigator = {};
 }
-navigator.product = 'ReactNative';
+
+// see https://github.com/facebook/react-native/issues/10881
+defineProperty(navigator, 'product', () => 'ReactNative');
```

Comment:
Can you add the (optional) eager argument to true? It's not worth having the overhead here of making this a lazy prop.

`defineProperty(navigator, 'product', () => 'ReactNative', true)`